thunder/tests/opinfos.py

```diff
@@ -2994,6 +2994,43 @@ def unbind_sample_generator(op, device, dtype, requires_grad, **kwargs):
 shape_ops.append(unbind_opinfo)
 
 
+def unfold_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        ((0,), 0, 0, 1),
```

Comment:
Let's add some tests for scalars, with different values of `size/step`. So that the line `max_size = 1 if a.ndim == 0 else a.shape[dim]` is justifiable.